index.js

```diff
@@ -27,16 +28,55 @@ const main = (fn, fp, data, opts) => {
 		});
 	}
 
-	const json = JSON.stringify(data, opts.replacer, opts.indent);
+	return fn(fp, data, opts);
+};
+
+const readFile = fp => {
+	return pify(fs.readFile)(fp, 'utf8')
+		.catch(err => {
+			if (err.code === 'ENOENT') {
+				return null;
+			}
+
+			throw err;
```

Comment:
Should we really throw here? I think it would be better to just silently use the default indent if the file doesn't exist. So just an empty catch handler here.